test/json.test.js

```diff
@@ -156,4 +156,92 @@ describe('parse.json(req, opts)', function() {
         .expect(200, done);
     });
   });
+
+  describe('JSON poisoning', function() {
+    it('remove inline __proto__ properties', function(done) {
+      const app = new koa();
+
+      app.use(async function(ctx) {
+        ctx.body = await parse.json(ctx, { returnRawBody: true });
+      });
+
+      const body = '{"foo": "bar", "__proto__": { "admin": true }}';
+
+      request(app.callback())
+        .post('/')
+        .type('json')
+        .send(body)
+        .expect(function(res) {
+          /* eslint no-proto: "off" */
+          res.body = { isAdmin: res.body.parsed.__proto__.admin };
+        })
+        .expect({ isAdmin: undefined })
+        .expect(200, done);
+    });
+
+    it('remove nested inline __proto__ properties', function(done) {
+      const app = new koa();
+
+      app.use(async function(ctx) {
+        ctx.body = await parse.json(ctx, { returnRawBody: true });
+      });
+
+      const body = '{"user": { "name": "virk", "__proto__": { "admin": true } }}';
+
+      request(app.callback())
+        .post('/')
+        .type('json')
+        .send(body)
+        .expect(function(res) {
+          /* eslint no-proto: "off" */
+          res.body = { isAdmin: res.body.parsed.user.__proto__.admin };
+        })
+        .expect({ isAdmin: undefined })
+        .expect(200, done);
+    });
+
+    it('error on inline __proto__ properties', function(done) {
+      const app = new koa();
+
+      app.use(async function(ctx) {
+        try {
```

Comment:
Should nothing be thrown, this test will basically hang instead of failing quickly. 

I tend to leverage a pattern of using a temp variable to store the err and then check it outside of the try/catch.